chain/network/src/peer_manager/network_state/mod.rs

```diff
@@ -386,72 +475,150 @@ impl NetworkState {
+                })
+            })
+            .await;
+            this.routing_table_view.add_local_edges(&edges);
+
+            let mut new_edges = match this
                 .routing_table_addr
-                .send(routing::actor::Message::AddVerifiedEdges { edges }.with_span_context())
+                .send(
+                    routing::actor::Message::AddVerifiedEdges { edges: edges.clone() }
+                        .with_span_context(),
+                )
                 .await
             {
-                Ok(routing::actor::Response::AddVerifiedEdgesResponse(mut edges)) => {
-                    this.config.event_sink.push(Event::EdgesVerified(edges.clone()));
-                    // Don't send tombstones during the initial time.
-                    // Most of the network is created during this time, which results
-                    // in us sending a lot of tombstones to peers.
-                    // Later, the amount of new edges is a lot smaller.
-                    if let Some(skip_tombstones_duration) = this.config.skip_tombstones {
-                        if clock.now() < this.start_time + skip_tombstones_duration {
-                            edges.retain(|e| e.edge_type() == EdgeState::Active);
-                            metrics::EDGE_TOMBSTONE_SENDING_SKIPPED.inc();
-                        }
-                    }
-                    this.tier2.broadcast_message(Arc::new(PeerMessage::SyncRoutingTable(
-                        RoutingTableUpdate::from_edges(edges),
-                    )));
+                Ok(routing::actor::Response::AddVerifiedEdges(new_edges)) => new_edges,
+                Ok(_) => panic!("expected AddVerifiedEdges"),
+                Err(err) => {
+                    tracing::error!(target:"network","routing::actor::Actor closed: {err}");
+                    return Ok(());
+                }
+            };
+            this.config.event_sink.push(Event::EdgesVerified(edges.clone()));
+            // Don't send tombstones during the initial time.
+            // Most of the network is created during this time, which results
+            // in us sending a lot of tombstones to peers.
+            // Later, the amount of new edges is a lot smaller.
+            if let Some(skip_tombstones_duration) = this.config.skip_tombstones {
+                if clock.now() < this.created_at + skip_tombstones_duration {
+                    new_edges.retain(|edge| edge.edge_type() == EdgeState::Active);
+                    metrics::EDGE_TOMBSTONE_SENDING_SKIPPED.inc();
                 }
-                _ => tracing::error!(target: "network", "expected AddVerifiedEdgesResponse"),
             }
-        });
+            // Broadcast new edges to all other peers.
+            this.broadcast_routing_table_update(Arc::new(RoutingTableUpdate::from_edges(
+                new_edges,
+            )))
+            .await;
+            if !ok {
+                return Err(ReasonForBan::InvalidEdge);
```

Comment:
In particlar, here: we already know we want to but the `peer`, but we somewhat artifically holding off broadcasting this bit of info.